### PR TITLE
Add CMake support

### PR DESCRIPTION
**Pull request by Shatur95**
Since qmake will soon become obsolete in favor of CMake, I suggest adding a rule for CMake :)

### CODE REVIEW

**Comment by Shatur95** [comment]
Also I suggest to remove `tr(...)` from the calculator example. Since when using CMake, these lines will be added to `*.ts` files.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,36 @@
+cmake_minimum_required(VERSION 3.12.4)
```
Comment (akallabeth) [inline CMakeLists.txt]: Are you sure such a high `CMake` version is required?

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    find_package(Qt5 COMPONENTS Core REQUIRED)
+endif()
+add_compile_definitions(QAPPLICATION_CLASS=${QAPPLICATION_CLASS})
+
+add_library(${PROJECT_NAME} STATIC
+    singleapplication.cpp
+    singleapplication_p.cpp
+    )
+
+target_link_libraries(${PROJECT_NAME} Qt5::Network)
```
Comment (akallabeth) [inline CMakeLists.txt]: Shouldn´t all dependencies for this be linked private?

File: examples/calculator/calculator.cpp [hunk 3]
```diff
@@ -82,27 +82,27 @@ Calculator::Calculator(QWidget *parent)
         digitButtons[i] = createButton(QString::number(i), SLOT(digitClicked()));
     }
 
-    Button *pointButton = createButton(tr("."), SLOT(pointClicked()));
```
Comment (akallabeth) [inline examples/calculator/calculator.cpp]: I don´t get why you remove the translation routines.
`CMake` automatically generates the ts files, but you still need to load the correct translation with `tr`

**Comment by akallabeth** [comment]
have modified your pull (see https://github.com/akallabeth/SingleApplication/tree/master) and changed a few things.
feel free to comment on the changes or take them for your pull if you like them.

File: examples/calculator/calculator.cpp (hunk 3 above)
Comment (Shatur95) [reply examples/calculator/calculator.cpp]: Yes, but CMake will generate .ts files with this lines in the project in which I will use this library.
And why the examples have the translation, if it is not used? It seems to me that the example should be minimal, which clearly demonstrates the library usage.

File: CMakeLists.txt (hunk 2 above)
Comment (Shatur95) [reply CMakeLists.txt]: Yes, you are right.

File: CMakeLists.txt (hunk 1 above)
Comment (Shatur95) [reply CMakeLists.txt]: No, sorry, it was generated by my IDE.

**Comment by Shatur95** [comment]
Thank you.

I think that this line is not needed, because ui is not used:
```
set(CMAKE_AUTOUIC ON)
```

Maybe set this option to `OFF` by default. In most cases, users do not want to build examples.
```
option(WITH_EXAMPLES "Build usage examples" ON)
```

If you agree that using the translation in the examples does not make sense, then this line can be deleted:
```
find_package(Qt5LinguistTools REQUIRED)
```
If not, then it should be used only in the assembly of examples.

```
find_package(Qt5LinguistTools REQUIRED)
```

Thus lines is also not needed.

```
include_directories(${Qt5_INCLUDE_DIRS})
add_definitions(${Qt5_DEFINITIONS})
```

`find_package` adds definitions and includes automatically.

**Comment by Shatur95** [comment]
I added the proposed changes, thanks!
But I made examples of separate projects as in qmake rules.

**Comment by jonaski** [comment]
Nice

**Pull request merged**